pkg/database/connection.go

```diff
@@ -97,3 +102,81 @@ func checkError(err error, logMessage string) {
 		klog.Error(logMessage, " ", err)
 	}
 }
+
+func (dao *DAO) UpsertCluster(resource model.Resource) {
+	var query string
+	var args []interface{}
+	data, _ := json.Marshal(resource.Properties)
+	clusterName := resource.Properties["name"].(string)
+
+	// Insert cluster node if cluster does not exist in the DB
+	if !dao.ClusterInDB(clusterName) {
+		args = []interface{}{resource.UID, "", string(data)}
+		klog.V(3).Infof("Cluster %s does not exist in DB, inserting it.", clusterName)
+		query = "INSERT into search.resources values($1,$2,$3)"
+	} else {
+		// Check if the cluster properties are up to date in the DB
+		if !dao.ClusterPropsUpToDate(clusterName, resource) {
+			args = []interface{}{resource.UID, string(data)}
+			klog.V(3).Infof("Cluster %s already exists in DB. Updating properties.", clusterName)
+			query = "UPDATE search.resources SET data=$2 WHERE uid=$1"
+		} else {
+			klog.V(3).Infof("Cluster %s already exists in DB and properties are up to date.", clusterName)
+			return
+		}
+	}
```

Comment:
This could be simplified with a single query using `INSERT ... ON CONFLICT ON CONSTRAINT uid ...`
Here are some examples.  https://www.postgresqltutorial.com/postgresql-upsert/

Reply:
Added